bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/VoiceManagerImpl.java

```diff
@@ -126,6 +126,7 @@ public class VoiceManagerImpl implements VoiceManager, ConfigOptionProvider {
     private final Map<String, String> defaultVoices = new HashMap<>();
 
     private Map<String, DialogProcessor> dialogProcessors = new HashMap<>();
+    private Map<String, DialogProcessor> singleDialogProcessors = new HashMap<>();
```

Comment:
You might want to use `ConcurrentHashMap` here since you modify the same map from different points in code which I believe can be called concurrently.